packages/phone-number-privacy/signer/src/web3/contracts.ts

```diff
@@ -16,23 +19,168 @@ export function getContractKit(): ContractKit {
   return contractKit
 }
 
+declare type InFunction<T extends any[], U> = (...params: T) => Promise<U>
+
+export async function meter<T extends any[], U>(
+  inFunction: InFunction<T, U>,
+  params: T,
+  onError: (err: any) => any, // TODO(Alec): Should this return U?
```

Comment:
It's return type should be `void` or `Promise<void>` I believe, based on reading the MDN docs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch